triton-vm/src/stark.rs

```diff
@@ -947,425 +1381,115 @@ impl Stark {
-        main_table
-            .trace_table_mut()
-            .axis_iter_mut(Axis(1))
-            .into_par_iter()
-            .for_each(|mut column| ntt(column.as_slice_mut().unwrap()));
-        aux_table
-            .trace_table_mut()
-            .axis_iter_mut(Axis(1))
-            .into_par_iter()
-            .for_each(|mut column| ntt(column.as_slice_mut().unwrap()));
-        profiler!(stop "restore original trace");
 
-        segmentification
-    }
-
-    /// Map a matrix whose columns represent the evaluation of a high-degree
-    /// polynomial f on all constituents of a partition of some large domain into
-    /// smaller cosets, to
-    /// 1. a matrix of segment codewords (on the FRI domain), and
-    /// 2. an array of matching segment polynomials,
-    ///
-    /// such that the segment polynomials correspond to the interleaving split of
-    /// the original high-degree polynomial.
-    ///
-    /// For example, let f(X) have degree M·N where N is the chosen domain's length.
-    /// Then the input is an N×M matrix representing the values of f(X) on the
-    /// chosen domain and its cosets:
-    ///
-    /// ```txt
-    /// ⎛  …          …   …           ⎞  ┬
-    /// ⎜ f(coset_0)  …  f(coset_M-1) ⎟ domain length
-    /// ⎝  …          …   …           ⎠  ┴
-    ///
-    /// ├───────── NUM_COSETS ────────┤
-    /// ```
-    ///
-    /// The `NUM_SEGMENTS` (=:`K`) produced segment polynomials are f_i(X) such that
-    /// f(X) = Σ_k x^k · f_k(X^K).
-    /// For example, for `K = 2`, this is f(X) = f_E(X²) + X·f_O(X²).
-    ///
-    /// The produced segment codewords are the segment polynomial's evaluations on
-    /// the FRI domain:
+        let fri_expansion_factor = 1 << log2_of_fri_expansion_factor;
+        let num_collinearity_checks = security_level / log2_of_fri_expansion_factor;
+
+        let num_out_of_domain_rows = 2;
+        let num_trace_randomizers = num_collinearity_checks
+            + num_out_of_domain_rows * x_field_element::EXTENSION_DEGREE
+            + NUM_QUOTIENT_SEGMENTS * x_field_element::EXTENSION_DEGREE;
+
+        Stark {
+            security_level,
+            fri_expansion_factor,
+            num_trace_randomizers,
+            num_collinearity_checks,
+        }
+    }
+
+    /// Prove the correctness of the given [Claim] using the given
+    /// [witness](AlgebraicExecutionTrace).
     ///
-    /// ```txt
-    /// ⎛  …            …   …             ⎞  ┬
-    /// ⎜ f_0(FRI_dom)  …  f_K-1(FRI_dom) ⎟ FRI domain length
-    /// ⎝  …            …   …             ⎠  ┴
+    /// This method should be the first option to consider for proving.
+    /// For more control over the proving process, see [`Prover`].
+    // This pass-through method guarantees fresh prover randomness at each call.
+    pub fn prove(
+        &self,
+        claim: &Claim,
+        aet: &AlgebraicExecutionTrace,
+    ) -> Result<Proof, ProvingError> {
+        Prover::new(*self).prove(claim, aet)
```

Comment:
As far as I can tell, this is the only line where something could have gone wrong, and I don't think anything is going wrong.